cli/shared/src/main/scala/com/lightbend/rp/reactivecli/process/dockercred.scala

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2017 Lightbend, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.lightbend.rp.reactivecli.process
+
+import com.lightbend.rp.reactivecli.concurrent._
+import com.lightbend.rp.reactivecli.docker.DockerCredentials
+import scala.collection.immutable.Seq
+import scala.concurrent.{Future, Await}
+import scala.concurrent.duration._
+import slogging._
+import argonaut._
+import Argonaut._
+
+object dockercred extends LazyLogging {
+  private def isAvailable(kind: String): Future[Boolean] = {
+    exec(s"docker-credential-$kind", "list").map(_._1 == 0)
+  }
+
+  private def chooseKind(): Future[String] = {
+    val kinds = Seq("osxkeychain", "wincred", "pass", "secretservice")
+    optionToFuture(kinds.find(kind => Await.result(isAvailable(kind), Duration.Inf)), "No docker credential helpers found")
+  }
```

Comment:
I'd probably do this recursively, maybe something like the below. Almost always, `Await.result` should be avoided, exception being tests. In this project we can get away with it because we're single threaded but we can still do better and keep the code usable from multithreaded contexts, e.g. JVM.

```scala
private def chooseKind(): Future[String] = {
  def step(ks: Seq[String]): Future[String] =
    if (ks.isEmpty)
      Future.failed(new NoSuchElementException("No docker credential helpers found")
    else
       isAvailable(ks.head).flatMap {
         case true => Future.successful(ks.head)
         case false => step(ks.tail)
       }

  step(Seq(...))
}
```